src/node_contextify.cc

```diff
@@ -511,6 +509,51 @@ class ContextifyContext {
 
     args.GetReturnValue().Set(ctx->sandbox()->GetPropertyNames());
   }
+
+
+  static void GlobalPropertyDefinerCallback(
+      Local<Name> property,
+      const PropertyDescriptor& desc,
+      const PropertyCallbackInfo<Value>& args) {
+    ContextifyContext* ctx;
+    ASSIGN_OR_RETURN_UNWRAP(&ctx, args.Data().As<Object>());
+
+    // Still initializing
+    if (ctx->context_.IsEmpty())
+      return;
+
+    Local<Context> context = ctx->context();
+
+    if (desc.has_value()) {
+      bool writable = desc.has_writable() ? desc.writable() : false;
+      PropertyDescriptor desc_copy(desc.value(), writable);
+      if (desc.has_enumerable()) {
+        desc_copy.set_enumerable(desc.enumerable());
+      }
+      if (desc.has_configurable()) {
+        desc_copy.set_configurable(desc.configurable());
+      }
+      Maybe<bool> result = ctx->sandbox()->DefineProperty(context, property, desc_copy);
```

Comment:
@fhinkel It feels wrong to me having to make manual copies of the `PropertyDescriptor`. Am I missing something?

Reply:
It's more or less correct. You have to know whether you want a value or accessor descriptor. enumerable, configurable, and DefineProp() is the same for both cases though.

Reply:
@targos Can you refactor the copying process into a new function? I suspect it might come handy in the future, and it'll also be nice to write the DefineProperty bit only once.

Reply:
I haven't tried playing around with it, but it does seem like one should be able to just pass in `desc` without making a copy. I'm sure there's an excellent reason why not tho :-)

Reply:
@jasnell I can't just pass `desc` because it's a const reference. `DefineProperty` wants a non-const reference.

Reply:
sigh... ok yeah, that's the kind of question that is easily answered by reading the docs ;-) That is unfortunate.